engine/runtime-compiler/src/main/java/org/enso/compiler/pass/lint/unusedimports/UsedSymbols.java

```diff
@@ -0,0 +1,70 @@
+package org.enso.compiler.pass.lint.unusedimports;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import org.enso.compiler.core.ir.module.scope.Import;
+import org.enso.pkg.QualifiedName;
+
+/** All the used symbols inside one module. */
+final class UsedSymbols {
+
+  private final Map<Import, Set<QualifiedName>> symbols;
+
+  private UsedSymbols(Map<Import, Set<QualifiedName>> symbols) {
+    this.symbols = symbols;
+  }
+
+  Set<QualifiedName> getUsedSymbolsForImport(Import importIr) {
+    if (!symbols.containsKey(importIr)) {
+      // Try to find the import based on location.
+      // It is possible that the import was replaced by a different instance
+      // with same location.
+      return symbols.entrySet().stream()
+          .filter(entry -> haveSameLocation(entry.getKey(), importIr))
+          .map(Entry::getValue)
+          .findFirst()
+          .orElse(Set.of());
+    }
+    return symbols.getOrDefault(importIr, Set.of());
+  }
+
+  private static boolean haveSameLocation(Import imp1, Import imp2) {
+    if (imp1.identifiedLocation() != null && imp2.identifiedLocation() != null) {
```

Comment:
Second incarnation of the same logic.

Reply:
Using single method in e44d0ccf558bb50c40bd081b8db1a92165383998